src/plugins/footnotes/index.ts

```diff
@@ -90,6 +102,108 @@ export const uncitedFootnoteWidget = () => () => {
   element.innerHTML = alertIcon
   return element
 }
+interface inlineFootnote {
+  node: InlineFootnoteNode | null
+  pos: number | null
+}
+
+const getInlineFootnote = (view: EditorView, id: string): inlineFootnote => {
+  let inlineFootnote: InlineFootnoteNode | null = null
+  let footnotePosition: number | null = null
+
+  view.state.doc.descendants((node, pos) => {
+    const footnote = node as InlineFootnoteNode
+
+    if (footnote.attrs.rids?.includes(id)) {
+      inlineFootnote = footnote
+
+      footnotePosition = pos
+      return false
+    }
+  })
+
+  return { node: inlineFootnote, pos: footnotePosition }
+}
+
+const deleteFootnoteWidget =
+  (
+    node: ManuscriptNode,
+    props: PluginProps,
+    footnoteType: string,
+    footnoteMessage: string,
+    id: string
+  ) =>
+  (view: EditorView, getPos: () => number | undefined) => {
+    const deleteBtn = document.createElement('span')
+    deleteBtn.className = 'delete-icon'
+    deleteBtn.innerHTML = deleteIcon
+
+    deleteBtn.addEventListener('click', () => {
+      const handleDelete = () => {
+        const tr = view.state.tr
+        const pos = getPos()
+
+        if (node.type === schema.nodes.table_element_footer) {
+          let isAllGeneralFootnotes = true
+
+          node.content.forEach((item) => {
+            if (item.type !== schema.nodes.paragraph) {
+              isAllGeneralFootnotes = false
+
+              return
+            }
+          })
+          if (isAllGeneralFootnotes && pos) {
```

Comment:
Replace that `isAllGeneralFootnotes` with `! getChildOfType(node, schema.nodes.footnotes_element, true)`

Reply:
done